HTTP 500 in `wp-json/batch` with specific arguments

bor0 opened the pull request:
Trac issue: https://core.trac.wordpress.org/ticket/63502

bor0 committed:
HTTP 500 in `wp-json/batch` with specific arguments

bor0 committed:
Coding standards

bor0 committed:
Coding standards

SirLouen reviewed (changes requested):
If you can review this patch with proposed changes
https://core.trac.wordpress.org/attachment/ticket/63502/63502.3.patch

bor0 commented:
> If you can review this patch with proposed changes

That seems to cause some phpunit issues.

```
bor0:~/dev/wp-svn$ svn revert -R .
Reverted 'src/wp-includes/rest-api/class-wp-rest-server.php'
Reverted 'tests/phpunit/tests/rest-api/rest-server.php'
bor0:~/dev/wp-svn$ vendor/bin/phpunit --filter REST_Server
Installing...
Running as single site... To run multisite, use -c tests/phpunit/multisite.xml
Not running ajax tests. To execute these, use --group ajax.
Not running ms-files tests. To execute these, use --group ms-files.
Not running external-http tests. To execute these, use --group external-http.
PHPUnit 9.6.23 by Sebastian Bergmann and contributors.

Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

...............................................................  63 / 122 ( 51%)
...........................................................     122 / 122 (100%)

Time: 00:03.295, Memory: 215.00 MB

OK (122 tests, 385 assertions)
bor0:~/dev/wp-svn$ patch -p0 < ~/Downloads/63502.3.patch 
patching file 'src/wp-includes/rest-api/class-wp-rest-server.php'
patching file 'tests/phpunit/tests/rest-api/rest-server.php'
bor0:~/dev/wp-svn$ vendor/bin/phpunit --filter REST_Server
Installing...
Running as single site... To run multisite, use -c tests/phpunit/multisite.xml
Not running ajax tests. To execute these, use --group ajax.
Not running ms-files tests. To execute these, use --group ms-files.
Not running external-http tests. To execute these, use --group external-http.
PHPUnit 9.6.23 by Sebastian Bergmann and contributors.

Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

...............................................................  63 / 123 ( 51%)
.....................................EEEEE....E.............    123 / 123 (100%)

Time: 00:03.341, Memory: 215.00 MB

There were 6 errors:

1) Tests_REST_Server::test_batch_v1_opt_in with data set "missing" (null, false)
Error: Call to undefined method WP_REST_Request::get_all_error_data()

/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:3410
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:214
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1827
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1292
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1125
/Users/bor0/dev/wp-svn/tests/phpunit/includes/spy-rest-server.php:71
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:586
/Users/bor0/dev/wp-svn/tests/phpunit/tests/rest-api/rest-server.php:2052

2) Tests_REST_Server::test_batch_v1_opt_in with data set "invalid type" (true, false)
Error: Call to undefined method WP_REST_Request::get_all_error_data()

/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:3410
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:214
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1827
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1292
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1125
/Users/bor0/dev/wp-svn/tests/phpunit/includes/spy-rest-server.php:71
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:586
/Users/bor0/dev/wp-svn/tests/phpunit/tests/rest-api/rest-server.php:2052

3) Tests_REST_Server::test_batch_v1_opt_in with data set "invalid type string" ('v1', false)
Error: Call to undefined method WP_REST_Request::get_all_error_data()

/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:3410
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:214
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1827
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1292
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1125
/Users/bor0/dev/wp-svn/tests/phpunit/includes/spy-rest-server.php:71
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:586
/Users/bor0/dev/wp-svn/tests/phpunit/tests/rest-api/rest-server.php:2052

4) Tests_REST_Server::test_batch_v1_opt_in with data set "wrong version" (array(true), false)
Error: Call to undefined method WP_REST_Request::get_all_error_data()

/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:3410
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:214
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1827
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1292
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1125
/Users/bor0/dev/wp-svn/tests/phpunit/includes/spy-rest-server.php:71
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:586
/Users/bor0/dev/wp-svn/tests/phpunit/tests/rest-api/rest-server.php:2052

5) Tests_REST_Server::test_batch_v1_opt_in with data set "false version" (array(false), false)
Error: Call to undefined method WP_REST_Request::get_all_error_data()

/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:3410
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:214
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1827
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1292
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1125
/Users/bor0/dev/wp-svn/tests/phpunit/includes/spy-rest-server.php:71
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:586
/Users/bor0/dev/wp-svn/tests/phpunit/tests/rest-api/rest-server.php:2052

6) Tests_REST_Server::test_batch_v1_partial_error
Error: Call to undefined method WP_REST_Request::get_all_error_data()

/Users/bor0/dev/wp-svn/src/wp-includes/rest-api.php:3410
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:214
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1827
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1292
/Users/bor0/dev/wp-svn/src/wp-includes/rest-api/class-wp-rest-server.php:1125
/Users/bor0/dev/wp-svn/tests/phpunit/includes/spy-rest-server.php:71
/Users/bor0/dev/wp-svn/tests/phpunit/tests/rest-api/rest-server.php:2281

ERRORS!
Tests: 123, Assertions: 371, Errors: 6.
bor0:~/dev/wp-svn$ 
```
